core/src/main/java/io/questdb/griffin/engine/groupby/SampleByInterpolateRecordCursorFactory.java

```diff
@@ -231,12 +241,28 @@ private class SampleByInterpolateRecordCursor extends VirtualFunctionSkewedSymbo
         private long prevSample = -1;
         private long rowId;
 
+        protected final Function offsetFunc;
+        protected final int offsetFuncPos;
+        protected final TimestampSampler timestampSampler;
+        protected final Function timezoneNameFunc;
+        protected final int timezoneNameFuncPos;
+        protected long fixedOffset;
+        protected long nextDstUtc;
```

Comment:
it looks this field is only assigned but never read? that's strange, isn't it? also: why is it protected and not private? some other fields could be private too.

Reply:
it seems it's entirely missing code for DST adjustment. and equivalent of this https://github.com/questdb/questdb/blob/9683d03251aa6284bf8d2de0d6d60884a477d893/core/src/main/java/io/questdb/griffin/engine/groupby/AbstractNoRecordSampleByCursor.java#L141